logstash-core/spec/logstash/pipeline_spec.rb

```diff
@@ -557,30 +576,47 @@ class TestPipeline < LogStash::Pipeline
 
       # Reset the metric store
       LogStash::Instrument::Collector.instance.clear
-    end
 
-    it "populates the differents core metrics" do
-      t = Thread.new { subject.run }
+      Thread.new { subject.run }
       # make sure we have received all the generated events
-      sleep 0.01 while dummyoutput.events.size < number_of_events
+      sleep 0.1 while dummyoutput.events.size < number_of_events
+    end
 
-      collected_metric = LogStash::Instrument::Collector.instance.snapshot_metric.metric_store.get_with_path("stats/events")
+    after :each do
+      subject.shutdown
+    end
 
+    it "populates the differents core metrics" do
+      collected_metric = LogStash::Instrument::Collector.instance.snapshot_metric.metric_store.get_with_path("stats/events")
```

Comment:
might be better to reduce this line size, we can probably use let var for the instance?